### DISCUSSION TITLE
Vite example

### DISCUSSION THREAD

**Discussion by tomhendra** [software-mansion/react-native-reanimated#5814] [Q&A]
Our project involves a universal Expo app and accompanying Vite Web app which is used for admin. 
We have a monorepo setup where we want to share components between apps. 
We are struggling to get Reanimated working with Vite. Does anyone have a working example please? 

Ideally we don't want to use Webpack as the example: https://github.com/software-mansion/react-native-reanimated/tree/main/WebExample since we are more familiar with Vite.

This is where we currently are: https://stackblitz.com/edit/vitejs-vite-zlodww?file=vite.config.ts

Any guidance would be very much appreciated!

**Comment by bajedev** [comment]
You don't seem to have the `@babel/core` dependency

**Reply by tomhendra** [reply]
Thanks for your input, but adding the `@babel/core` dependency doesn't seem to solve the problem.

**Reply by bajedev** [reply]
https://github.com/software-mansion/react-native-reanimated/assets/32736293/e538ad07-d05a-4366-aa4c-64ce9a7aebc2

Here's what happens after I add babel core

- At first nothing changes
- When I remove width from the styles then put it back everything starts to work even if I make changes to the animation

**Reply by bajedev** [reply]
You can also simplify the vite config by using https://github.com/Bram-dc/vite-plugin-react-native-web

**Comment by nkoynov** [comment]
When using `react-native-reanimated` on the web with Strict Mode enabled, the issue arises because react-native-reanimated's current version isn't fully compatible with React's Strict Mode on web platforms. You can disable the Strict Mode and it will work.

Here is more information:
https://github.com/software-mansion/react-native-reanimated/issues/6264

**Reply by nkoynov** [reply]
https://github.com/user-attachments/assets/06ba9dd4-6a70-4acc-a439-6406f66bdebc

**Reply by bajedev** [reply]
Awesome

**Comment by dannyhw** [comment]
Hey currently also trying to setup vite with reanimated and the problem I get which also reproduces in the stackblitz example is that on the built version I get this error:
```
index-D4qkjY-g.js:53 Uncaught TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at dl (index-D4qkjY-g.js:53:6365)
    at index-D4qkjY-g.js:53:48960
    at Array.forEach (<anonymous>)
    at Dc (index-D4qkjY-g.js:53:48934)
    at Object.a [as worklet] (index-D4qkjY-g.js:53:51069)
    at o (index-D4qkjY-g.js:53:14190)
```

![image](https://github.com/user-attachments/assets/1a9900b5-db52-4999-bc9d-70befbe4d44f)

**Reply by dannyhw** [reply]
Created a related issue here since I haven't been able to get the build working 
https://github.com/software-mansion/react-native-reanimated/issues/6775

**Reply by dannyhw** [reply]
I found my own solution eventually after digging through expo's babel preset

In order to make it work I had to add the babel plugin for transforming commonjs

```
import babel from 'vite-plugin-babel';
// ... stuff
babel({
   include: [/node_modules\/(react-native|@react-native)/],
  // ... stuff
  plugins: [
    [
      // this is a fix for reanimated not working in production
      '@babel/plugin-transform-modules-commonjs',
      {
        strict: false,
        strictMode: false, // prevent "use strict" injections
        allowTopLevelThis: true, // dont rewrite global `this` -> `undefined`
      },
    ],
    ...otherplugins
  ],
})
```

**Comment by ayuleul** [comment]
Based on @dannyhw's answer, the following solution works for us:
```
import babel from 'vite-plugin-babel';

export default defineConfig({
  plugins: [
    babel({
      babelConfig: {
        include: [/node_modules\/(react-native|@react-native)/],
        plugins: [
          [
            '@babel/plugin-transform-modules-commonjs',
            {
              strict: false,
              strictMode: false,
              allowTopLevelThis: true,
            },
          ],
        ],
      },
    }),
    ...otherPlugins,
  ],
  ...other,
});
```
